### PR TITLE
Feat: added replace under the string manipulation

### PR DESCRIPTION
**Pull request by GIHAA**
#37

### CODE REVIEW

File: test/modules/string/replace_test.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import 'package:flodash/flodash.dart' as flodash;
+
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('replace', () {
```
Comment (Akalanka47000) [inline test/modules/string/replace_test.dart:6]: This group isn't needed ne, all of the tests inside this file will be related to the replace function only. It's adding a little bit of additional nesting, what do you think ?

File: test/modules/string/replace_test.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    test('replace should replace all occurrences of pattern in the string', () {
+      String input = 'Hello, Hello, Hello!';
+      String expected = 'Hi, Hi, Hi!'; // Update the expected result
```
Comment (Akalanka47000) [inline test/modules/string/replace_test.dart:9]: This comment can be safely removed I guess, the variable name is self explanatory.

File: test/modules/string/replace_test.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      String actual = flodash.replace(input, 'Hello', 'Hi');
+      expect(actual, expected);
+    });
+
+    test('should replace pattern with an empty string', () {
+      String input = 'Hello, World!';
+      String pattern = 'o';
+
```
Comment (Akalanka47000) [inline test/modules/string/replace_test.dart:17]: Let's not leave additional spacing in between lines

File: lib/modules/string/replace.dart [hunk 4]
```diff
@@ -0,0 +1,4 @@
+String replace(String string, dynamic pattern, dynamic replacement) {
+  Pattern patternObj = pattern is String ? RegExp(pattern) : pattern;
```
Comment (Akalanka47000) [inline lib/modules/string/replace.dart:2]: This condition might be safer if we evaluate it the other way around as follows:-

```dart
Pattern patternObj = pattern is RegExp ? pattern : RegExp(pattern.toString());
```

**Review by Akalanka47000 — CHANGES_REQUESTED** [review]

File: test/modules/string/replace_test.dart (hunk 1 above)
Comment (GIHAA) [reply test/modules/string/replace_test.dart]: yes i will remove it

**Review by Akalanka47000 — APPROVED** [review]

**Pull request merged**